### ISSUE TITLE
zpool attach, replace /resilvering currently doesn't work reliably

### ISSUE THREAD

**Issue by JMoVS** [openzfsonosx/zfs#693]
Resilvering after replacing is borked right now, attempts to attach a disk, scrub or resilver afterwards also fail, to reproduce:

```
zpool create tank disk0
zpool attach tank disk0 disk1
zpool replace tank disk0 disk2
rsync ./ /Volumes/tank/
```

checksum errors, IO errors etc occur

**Comment by lundman** [comment]
```
9a0c2f8c7b4ad5445485224999b0395b0c2855dd is the first bad commit
commit 9a0c2f8c7b4ad5445485224999b0395b0c2855dd
Author: Jorgen Lundman <lundman@lundman.net>
Date:   Tue Feb 19 14:36:50 2019 +0900

    8423 Implement large_dnode pool feature (#684)
```

**Comment by lundman** [comment]
The rsync command itself will spew errors:

```
rsync: mkstemp "/Volumes/tank/include/.zfs_comutil.h~.YC8uPd" failed: Malformed Mach-o file (88)
```
(Error 88 is ECKSUM)

and I simply copy the build zfs tree over, roughly 1.3G of data.

**Comment by JMoVS** [comment]
Okay, Did you revert/fix the problematic commit?

**Comment by lundman** [comment]
It is interesting to note that the replace command has to be the original `disk0`. Replacing `disk1` in the above example does not trigger the problem.

**Comment by JMoVS** [comment]
it gets weirder every day

**Comment by JMoVS** [comment]
had a weird scrubbing issue after reboot, but may also be because the kextcache might have contained the old broken master and not current one

**Comment by lundman** [comment]
Ok, turns out to be this code, merged wrong from conflict:
![Screen Shot 2019-03-11 at 1124 44 (2)](https://user-images.githubusercontent.com/637168/54097664-81300580-43f4-11e9-9f6f-3c2f356b48d7.png)

will be fixed in the up coming commit d721890e38c8f17242eada3423027af2f6454443 once the zfs-tester run completes.

**Comment by JMoVS** [comment]
what did the tester run say?

**Comment by lundman** [comment]
It nearly completed but ran out of diskspace - running again next time I'm away from desktop for 2 hours.

**Comment by lundman** [comment]
```
FAIL	 228
PASS	 664

Running Time:	03:19:58
Percent passed:	74.4%
```

**Comment by lundman** [comment]
Fixed with d721890
